test/Microsoft.IdentityModel.Tokens.Tests/Validation/AlgorithmValidationResultTests.cs

```diff
@@ -0,0 +1,150 @@
+                        },
+                        AlgorithmValidationResult = new AlgorithmValidationResult(SecurityAlgorithms.Sha256)
+                    },
+                    new AlgorithmTheoryData
+                    {
+                        TestId = "Invalid_ValidateAlgorithmUsingDelegate",
+                        ExpectedException = ExpectedException.SecurityTokenInvalidAlgorithmException("IDX10697:"),
+                        Algorithm = SecurityAlgorithms.Sha256,
+                        SecurityKey = securityKey,
+                        SecurityToken = null,
+                        ValidationParameters = new TokenValidationParameters
+                        {
+                            AlgorithmValidator = (algorithm, securityKey, securityToken, validationParameters) => false
+                        },
+                        AlgorithmValidationResult = new AlgorithmValidationResult(
+                            SecurityAlgorithms.Sha256,
+                            ValidationFailureType.AlgorithmValidationFailed,
+                            new ExceptionDetail(
+                                new MessageDetail(
+                                    LogMessages.IDX10697,
+                                    LogHelper.MarkAsNonPII(SecurityAlgorithms.Sha256),
+                                    securityKey),
+                                typeof(SecurityTokenInvalidAlgorithmException),
+                                new StackFrame(true)))
+                    },
+                    new AlgorithmTheoryData
+                    {
+                        TestId = "Invalid_ValidateAlgorithmNotAValidAlgorithm",
+                        ExpectedException = ExpectedException.SecurityTokenInvalidAlgorithmException("IDX10696:"),
+                        Algorithm = SecurityAlgorithms.Sha256,
+                        SecurityKey = securityKey,
+                        SecurityToken = null,
+                        ValidationParameters = new TokenValidationParameters
+                        {
+                            ValidAlgorithms = new[] { SecurityAlgorithms.HmacSha256 }
+                        },
+                        AlgorithmValidationResult = new AlgorithmValidationResult(
+                            SecurityAlgorithms.Sha256,
+                            ValidationFailureType.AlgorithmValidationFailed,
+                            new ExceptionDetail(
+                                new MessageDetail(
+                                    LogMessages.IDX10696,
+                                    LogHelper.MarkAsNonPII(SecurityAlgorithms.Sha256),
+                                    securityKey),
+                                typeof(SecurityTokenInvalidAlgorithmException),
+                                new StackFrame(true)))
+                    },
+                    new AlgorithmTheoryData
```

Comment:
Can we add a test case for the Valid scenario where there's no `ValidAlgorithms` assigned to our `ValidationParameters`?
Just to highlight all the success/failure scenarios.

Reply:
Not sure I follow, if we have no `ValidAlgorithms` assigned to our `ValidationParameters` then, it would always be an invalid scenario and should fail so, the scenario above in line 99 should cover that right?

Reply:
Added test case. Thanks Ignacio!